src/Microsoft.AspNetCore.Mvc.Razor/Compilation/ICompilationService.cs

```diff
@@ -1,13 +1,15 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using Microsoft.AspNetCore.Razor.Language;
 
 namespace Microsoft.AspNetCore.Mvc.Razor.Compilation
 {
     /// <summary>
     /// Provides methods for compilation of a Razor page.
     /// </summary>
+    [Obsolete("This type will be removed in a future version. Use RazorCompiler instead.")]
```

Comment:
I'll check with @Eilon if we can remove the type entirely or mark it as obsolete for now and remove it in a later version.